src/storage/exec/IndexScanNode.h

```diff
@@ -100,56 +100,62 @@ class IndexScanNode : public RelNode<T> {
         std::string prefix;
         prefix.append(IndexKeyUtils::indexPrefix(partId, indexId_));
         for (auto& col : columnHints_) {
-            auto iter = std::find_if(fields.begin(), fields.end(),
-                                 [col](const auto& field) {
-                                     return col.get_column_name() == field.get_name();
-                                 });
+            auto iter = std::find_if(fields.begin(), fields.end(), [col](const auto& field) {
+                return col.get_column_name() == field.get_name();
+            });
             if (iter == fields.end()) {
                 VLOG(3) << "Field " << col.get_column_name() << " not found ";
                 return Status::Error("Field not found");
             }
-            auto type = iter->get_type().get_type();
-            if (IndexKeyUtils::toValueType(type) == Value::Type::STRING) {
-                if (!iter->get_type().__isset.type_length) {
-                    return Status::Error("String property index has not set prefix length.");
-                }
-                auto len = *iter->get_type().get_type_length();
-                prefix.append(IndexKeyUtils::encodeValue(col.get_begin_value(), len));
-            } else {
-                prefix.append(IndexKeyUtils::encodeValue(col.get_begin_value()));
+            auto type = IndexKeyUtils::toValueType(iter->type.type);
+            if (type == Value::Type::STRING && !iter->type.__isset.type_length) {
+                return Status::Error("String property index has not set prefix length.");
             }
+            prefix.append(encodeValue(col.begin_value, type, iter->type.get_type_length()));
```

Comment:
When col  is a not string type, ColumnTypeDef::type_length is not set. 
So the original code is reasonable.

Reply:
Got. In `IndexScanNode::encodeValue `， use it when the pointer `iter->type.get_type_length()` is valid

Reply:
Yeah, it is a pointer.